apollo-runtime/src/main/java/com/apollographql/apollo/ApolloClient.java

```diff
@@ -615,8 +622,14 @@ public ApolloClient build() {
       SubscriptionManager subscriptionManager = new NoOpSubscriptionManager();
       Optional<SubscriptionTransport.Factory> subscriptionTransportFactory = this.subscriptionTransportFactory;
       if (subscriptionTransportFactory.isPresent()) {
+        final ApolloStore finalApolloStore = apolloStore;
         subscriptionManager = new RealSubscriptionManager(scalarTypeAdapters, subscriptionTransportFactory.get(),
-            subscriptionConnectionParams, dispatcher, subscriptionHeartbeatTimeout);
+            subscriptionConnectionParams, dispatcher, subscriptionHeartbeatTimeout,
+            new Supplier<ResponseNormalizer<Map<String, Object>>>() {
+              @Override public ResponseNormalizer<Map<String, Object>> get() {
+                return finalApolloStore.networkResponseNormalizer();
```

Comment:
Why does this have to be lazy ?

Reply:
The issue that instance of `ResponseNormalizer` is stateful, means we need to create a new instance everytime we parse response. So it's not lazy but rather a factory to be called whenever we parse new response from the server.